src/Symfony/Component/Asset/Preload/PreloadManager.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\Preload;
+
+/**
+ * Manages preload HTTP headers.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class PreloadManager implements PreloadManagerInterface
```

Comment:
> My root issue is that final here would be only forbidding use cases while providing no benefit for us.

There are 2 benefits: 
* changing the methods is less risky, see for example: #11708, it was rejected just because of potential BC breaking;
* discourage "decoration by inheritance" when you have an interface with the same methods: in most cases there is no sane reason to do that, it is probably design mistake.

> When an interface covers the said methods, the contract is already enforced and inheritance should not be prevented.

I don't see connection between contract enforcing and "inheritance should not be prevented". I'd say the opposite. See similar opinion, for example: http://ocramius.github.io/blog/when-to-declare-classes-final/ /cc @Ocramius